### PR TITLE
[3015] Handle lane selection and editing in Deck

### LINKED ISSUE #3015: Handle Lane editing and selection in the Deck representation
* [ ] **I have checked that this feature has not yet been suggested by someone else.**

### Expected behavior
Lane should be selectable and editable in the same way as card. 
...

### Cuttings backs

...

### Potential side effects

...

### Describe alternatives you've considered

...

### PR DESCRIPTION
**Pull request by florianbarbin**
See bug #3015 

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?

- Create a Blank Studio and init a Deck View by using the stereotype
- Create Task project from the template
- Create Deck representation on the "Daily Dev Project"
- Select a Lane : the underlying tag is selected and visible in the details view
- Select a daily tag: the corresponding lane is selected
- Select a Lane and press F2: the focus is on the title field
- Edit the title: the tag suffix is modified in the model

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/deck/frontend/sirius-components-deck/src/representation/DeckRepresentation.tsx [hunk 1]
```diff
@@ -247,6 +304,23 @@ export const DeckRepresentation = ({ editingContextId, representationId }: Repre
...
+  const handleLaneClicked = (laneId: string) => {
+    if (deck) {
+      const lane: GQLLane | undefined = findLaneById(deck, laneId);
+      if (lane) {
+        setSelection({
+          entries: [
+            {
+              id: lane.targetObjectId,
+              kind: '',
+              label: '',
```
Comment (frouene) [inline packages/deck/frontend/sirius-components-deck/src/representation/DeckRepresentation.tsx]: Even if it's no use today, wouldn't it be better to set a kind and a label?
`kind: 'lane'`
`label: lane.label`

File: packages/deck/frontend/sirius-components-deck/src/laneHeader/DeckLaneHeader.types.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+export interface DeckLaneHeaderProps {
+  updateTitle: (value: string) => void;
+  editLaneTitle: boolean;
+  label: string;
+  title: string;
+  t: (value: string) => string;
```
Comment (frouene) [inline packages/deck/frontend/sirius-components-deck/src/laneHeader/DeckLaneHeader.types.ts]: could you use an explicite variable name.

File: integration-tests/cypress/e2e/project/deck/deck-lane.cy.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    deck.getLane('Monday').should('have.css', 'border', '2px solid rgb(190, 26, 120)');
+    deck.getLane('Wednesday').click();
+
+    // We verify that when selecting a card, the task is selected too in the model explorer.
+    explorer.getSelectedTreeItems();
+    explorer.getSelectedTreeItems().should('have.length', 1);
+    explorer.getSelectedTreeItems().contains('daily::Wednesday').should('exist');
+  });
+
+  it.only('We verify the lane direct edit.', () => {
```
Comment (frouene) [inline integration-tests/cypress/e2e/project/deck/deck-lane.cy.ts]: `only` !

**Review by frouene — CHANGES_REQUESTED** [review]
When testing selection and direct edit, it's working.
There is however, a strange behavior with the `F2`. It's working the first time but if the lane stay selected (for example after `escape`) then a new `F2` doesn't trigger direct edit.

File: integration-tests/cypress/e2e/project/deck/deck-lane.cy.ts (hunk 3 above)
Comment (florianbarbin) [reply integration-tests/cypress/e2e/project/deck/deck-lane.cy.ts]: done

File: packages/deck/frontend/sirius-components-deck/src/laneHeader/DeckLaneHeader.types.ts (hunk 2 above)
Comment (florianbarbin) [reply packages/deck/frontend/sirius-components-deck/src/laneHeader/DeckLaneHeader.types.ts]: done

File: packages/deck/frontend/sirius-components-deck/src/representation/DeckRepresentation.tsx (hunk 1 above)
Comment (florianbarbin) [reply packages/deck/frontend/sirius-components-deck/src/representation/DeckRepresentation.tsx]: done

**Comment by florianbarbin** [comment]
> When testing selection and direct edit, it's working. There is however, a strange behavior with the `F2`. It's working the first time but if the lane stay selected (for example after `escape`) then a new `F2` doesn't trigger direct edit.

I raised an issue for this point: https://github.com/eclipse-sirius/sirius-web/issues/3070

**Review by frouene — APPROVED** [review]

**Pull request merged**